server/clcachesrv.py

```diff
@@ -36,8 +36,11 @@ def getFileHash(self, path):
         return hashsum
 
     def _startWatching(self, dirname):
-        ev = pyuv.fs.FSEvent(self._loop)
-        ev.start(dirname, 0, self._onPathChange)
+        self.ev = pyuv.fs.FSEvent(self._loop)
+        self.ev.start(dirname, 0, self._onPathChange)
+        
+    def __del__(self):
+        self.ev.stop()
```

Comment:
`HashCache` is shared between different paths. But right now you stop to listen only last one - leak a still in place.
I think it makes sense to fill new PR and address it a little bit different way:
   Start new handle -> put result object to list -> close all of them in the loop.